src/main/java/core/Game.java

```diff
@@ -556,13 +556,16 @@ public static Game runOne(GameType gameToPlay, List<AbstractPlayer> players, lon
 
             GUI frame = new GUI();
             AbstractGUIManager gui = null;
+            boolean runWithVisuals = ac != null;
+
             if (ac != null) {
                 // Create GUI (null if not implemented; running without visuals)
                 GamePanel gamePanel = new GamePanel();
                 frame.setContentPane(gamePanel);
                 gui = gameToPlay.createGUIManager(gamePanel, game, ac);
             }
-            frame.setFrameProperties();
+
+            frame.setFrameProperties(runWithVisuals);
```

Comment:
Unnecessary, revert. If running with visuals (ac != null), then gui will not be null and the frame should always be drawn on, so it should just be always visible.

Reply:
done in part 1